src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/commissioner/CommissionerWorker.java

```diff
@@ -143,16 +143,7 @@ public CommissionerWorker(@NonNull Context context, @NonNull WorkerParameters wo
   }
 
   String getPskd() {
-    final int pskdLength = 27 / 5 + 1;
-    StringBuilder str = new StringBuilder();
-
-    for (int i = 0; i < pskdLength; ++i) {
-      str.append(
-          "0123456789ABCDEFGHJKLMNPRSTUVWXY"
-              .charAt(((int) deviceInfo.getSetupPinCode() >> (i * 5)) & ((1 << 5) - 1)));
-    }
-
-    return str.toString();
+    return String.format("%07u", deviceInfo.getSetupPinCode());
```

Comment:
I believe you want 9 digits.

2 ** 27 - 1 = 134217727

Reply:
you are right, I was counting in Hex but printing in decimal! Which one would you prefer? Hex result in shorter PSKd.

Reply:
Looks like Apple enforce the setup pin code to be decimal digits. Updated to convert to PSKd as 9 decimal digits.

Reply:
For the purposes of Thread PSKd, I do not think it really matters how we encode the string. The CHIP manual pairing code has its own encoding that includes other fields, so we may want to extract those 27 bits out of the manual pairing code anyway. I don't have a strong preference between base32, hex, and decimal, although base32 does result in a smaller string.

Reply:
@jwhui Thanks for comments. We had a small discussion about the format and agreed with 9 bytes decimal string for now.